planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp

```diff
@@ -3300,12 +3309,15 @@ void AvoidanceModule::insertWaitPoint(
     return;
   }
 
-  const auto stop_distance = helper_.getMildDecelDistance(0.0);
-  if (stop_distance) {
-    const auto insert_distance = std::max(data.to_stop_line, *stop_distance);
+  if (data.stop_target_object.get().is_stoppable) {
     utils::avoidance::insertDecelPoint(
-      getEgoPosition(), insert_distance, 0.0, shifted_path.path, stop_pose_);
+      getEgoPosition(), data.to_stop_line, 0.0, shifted_path.path, stop_pose_);
+    return;
   }
+
+  const auto stop_distance = helper_.getMildDecelDistance(0.0);
```

Comment:
// If the object cannot be stopped for, calculate a "mild" deceleration distance
  // and insert a deceleration point at that distance

Reply:
Add comment line in https://github.com/autowarefoundation/autoware.universe/pull/3978/commits/0d099f4f4baaa6ff9134f77160612075a7b5c162.